### PR TITLE
Added error checks for fscache saving

### PR DESCRIPTION
**Pull request by marcelofg55**
I found out about these after I started godot on a new OS install with root user by mistake and the .godot config dir got root permissions, so then when I started godot with my regular user it was crashing while trying to write the fscache.

### CODE REVIEW

File: editor/editor_settings.cpp [hunk 1]
```diff
@@ -856,7 +856,7 @@ void EditorSettings::save() {
 	Error err = ResourceSaver::save(singleton->config_file_path, singleton);
 
 	if (err != OK) {
-		ERR_PRINT("Can't Save!");
+		ERR_PRINTS("Can't Save! " + singleton->config_file_path);
```
Comment (akien-mga) [inline editor/editor_settings.cpp]: Could you rephrase to `"Error saving editor settings to " + <str>`?

**Comment by akien-mga** [comment]
Do you mean that when starting Godot has root, it still wrote its data in the non-root user's $HOME? I'd expect the root Godot to write its data in `/root/.config/godot`, which is then not access by the non-root user using `/home/<username>/.config/godot`.

**Comment by marcelofg55** [comment]
> Do you mean that when starting Godot has root, it still wrote its data in the non-root user's $HOME? I'd expect the root Godot to write its data in /root/.config/godot, which is then not access by the non-root user using /home/<username>/.config/godot.

I used sudo -s before actually running godot, so if you run godot like that if will still write to the user config dir.

File: editor/editor_settings.cpp (hunk 1 above)
Comment (marcelofg55) [reply editor/editor_settings.cpp]: Yeah, sure, I'll do that in a min.

**Comment by marcelofg55** [comment]
@akien-mga Just pushed the change on that string you asked :)

**Comment by hpvb** [comment]
Cherry picked into 3.0.3
